### PR TITLE
read Jackalope state for crashes

### PR DESCRIPTION
**Pull request by LivInAbsurdism**
Adds: 
- `connection_query` function that makes a GenServer call to get the state of `Jackalope.Session`
- GenServer callback that returns the connection_status

### CODE REVIEW

File: lib/jackalope/session.ex [hunk 1]
```diff
@@ -43,6 +43,11 @@ defmodule Jackalope.Session do
     GenServer.cast(__MODULE__, {:report_connection_status, status})
   end
 
+  @spec connection_query() :: :online | :offline
+  def connection_query() do
```
Comment (jjcarstens) [inline lib/jackalope/session.ex:47]:
```suggestion
  def connection_status() do
```
Can we have this match the key of the state it is returning?

Comment (LivInAbsurdism) [reply lib/jackalope/session.ex:47]: Just making sure I understand, we want to match the key in the args?

Comment (LivInAbsurdism) [reply lib/jackalope/session.ex]: updated

File: lib/jackalope.ex [hunk 2]
```diff
@@ -194,6 +194,12 @@ defmodule Jackalope do
              ]
   defdelegate publish(topic, payload, opts \\ []), to: Jackalope.Session
 
+  @doc """
+  For returning the `connection_status`.
```
Comment (jjcarstens) [inline lib/jackalope.ex:198]:
```suggestion
  Get the current MQTT connection status
```

File: lib/jackalope/session.ex [hunk 3]
```diff
@@ -43,6 +43,13 @@ defmodule Jackalope.Session do
     GenServer.cast(__MODULE__, {:report_connection_status, status})
   end
 
+  @spec connection_status() :: :online | :offline
+  def connection_status() do
+    GenServer.call(__MODULE__, :connection_status)
+  catch
+    _, _ -> false
```
Comment (jjcarstens) [inline lib/jackalope/session.ex:50]:
```suggestion
    _, _ -> :offline
```

This should prob be `:offline` to match the spec

Comment (LivInAbsurdism) [reply lib/jackalope/session.ex:50]: updated.

**Review by jjcarstens — APPROVED** [review]
🎉

**Review by jfcloutier — APPROVED** [review]

**Pull request merged**